apps/dashboard/src/app/knowledge-box/knowledge-box/knowledge-box.component.ts

```diff
@@ -1,22 +1,34 @@
-import { Component, OnInit } from '@angular/core';
-import { forkJoin, take } from 'rxjs';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { distinctUntilKeyChanged, forkJoin, Subject, switchMap, take, takeUntil } from 'rxjs';
 import { SisModalService } from '@nuclia/sistema';
 import { FeaturesService, SDKService } from '@flaps/core';
 import { GETTING_STARTED_DONE_KEY } from '@nuclia/user';
 import { GettingStartedComponent } from '../../onboarding/getting-started/getting-started.component';
 import { WelcomeInExistingKBComponent } from '../../onboarding/welcome-in-existing-kb/welcome-in-existing-kb.component';
+import { SearchWidgetStorageService } from '@flaps/common';
 
 @Component({
   template: '<router-outlet></router-outlet>',
 })
-export class KnowledgeBoxComponent implements OnInit {
+export class KnowledgeBoxComponent implements OnInit, OnDestroy {
+  unsubscribeAll = new Subject<void>();
```

Comment:
usually this should be a private property

Reply:
Fixed